### PR TITLE
run rustfmt on map.rs in libcollections/btree folder

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

File: src/libcollections/btree/map.rs [hunk 1]
```diff
@@ -129,35 +129,38 @@ use self::Entry::*;
...
+    length: usize,
 }
 
 impl<K, V> Drop for BTreeMap<K, V> {
     #[unsafe_destructor_blind_to_params]
     fn drop(&mut self) {
         unsafe {
-            for _ in ptr::read(self).into_iter() { }
+            for _ in ptr::read(self).into_iter() {
+            }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Err... This looks weird and against standard Rust style to me.

File: src/libcollections/btree/map.rs [hunk 2]
```diff
@@ -244,7 +249,8 @@ impl<K, Q: ?Sized> super::Recover<Q> for BTreeMap<K, ()>
                     handle: handle,
                     length: &mut self.length,
                     _marker: PhantomData,
-                }.insert(());
+                }
+                .insert(());
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: This is debatable, but I think things are clearer without the extra line.

File: src/libcollections/btree/map.rs [hunk 3]
```diff
@@ -311,15 +317,13 @@ pub struct RangeMut<'a, K: 'a, V: 'a> {
 pub enum Entry<'a, K: 'a, V: 'a> {
     /// A vacant Entry
     #[stable(feature = "rust1", since = "1.0.0")]
-    Vacant(
-        #[stable(feature = "rust1", since = "1.0.0")] VacantEntry<'a, K, V>
-    ),
+    Vacant(#[stable(feature = "rust1", since = "1.0.0")]
+           VacantEntry<'a, K, V>),
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Why not just put everything on one line?

File: src/libcollections/btree/map.rs [hunk 4]
```diff
@@ -311,15 +317,13 @@ pub struct RangeMut<'a, K: 'a, V: 'a> {
...
 
     /// An occupied Entry
     #[stable(feature = "rust1", since = "1.0.0")]
-    Occupied(
-        #[stable(feature = "rust1", since = "1.0.0")] OccupiedEntry<'a, K, V>
-    ),
+    Occupied(#[stable(feature = "rust1", since = "1.0.0")]
+             OccupiedEntry<'a, K, V>),
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Ditto.

File: src/libcollections/btree/map.rs [hunk 5]
```diff
@@ -349,7 +349,7 @@ pub struct OccupiedEntry<'a, K: 'a, V: 'a> {
 }
 
 // An iterator for merging two sorted sequences into one
-struct MergeIter<K, V, I: Iterator<Item=(K, V)>> {
+struct MergeIter<K, V, I: Iterator<Item = (K, V)>> {
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: I don't mind this, but it surprises me - most of the code I've seen in the Rust codebase does not use spaces around `=` in bounds on associated types.

File: src/libcollections/btree/map.rs [hunk 6]
```diff
@@ -628,47 +641,63 @@ impl<K: Ord, V> BTreeMap<K, V> {
...
+                match search::search_tree(self.root.as_ref(), key) {
+                    Found(kv_handle) => {
+                        match kv_handle.right_edge().force() {
+                            Leaf(bottom) => bottom,
+                            Internal(internal) => first_leaf_edge(internal.descend()),
+                        }
+                    }
+                    GoDown(bottom) => bottom,
+                }
+            }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: I don't see the advantage of these extra braces.

File: src/libcollections/btree/map.rs [hunk 7]
```diff
@@ -628,47 +641,63 @@ impl<K: Ord, V> BTreeMap<K, V> {
...
+                match search::search_tree(self.root.as_ref(), key) {
+                    Found(kv_handle) => {
+                        match kv_handle.left_edge().force() {
+                            Leaf(bottom) => bottom,
+                            Internal(internal) => last_leaf_edge(internal.descend()),
+                        }
+                    }
+                    GoDown(bottom) => bottom,
+                }
+            }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Ditto

File: src/libcollections/btree/map.rs [hunk 8]
```diff
@@ -704,51 +733,67 @@ impl<K: Ord, V> BTreeMap<K, V> {
...
+                match search::search_tree(root1, key) {
+                    Found(kv_handle) => {
+                        match kv_handle.right_edge().force() {
+                            Leaf(bottom) => bottom,
+                            Internal(internal) => first_leaf_edge(internal.descend()),
+                        }
+                    }
+                    GoDown(bottom) => bottom,
+                }
+            }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Ditto

File: src/libcollections/btree/map.rs [hunk 9]
```diff
@@ -704,51 +733,67 @@ impl<K: Ord, V> BTreeMap<K, V> {
...
+                match search::search_tree(root2, key) {
+                    Found(kv_handle) => {
+                        match kv_handle.left_edge().force() {
+                            Leaf(bottom) => bottom,
+                            Internal(internal) => last_leaf_edge(internal.descend()),
+                        }
+                    }
+                    GoDown(bottom) => bottom,
+                }
+            }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Ditto

File: src/libcollections/btree/map.rs [hunk 10]
```diff
@@ -773,21 +818,25 @@ impl<K: Ord, V> BTreeMap<K, V> {
...
+                })
+            }
+            GoDown(handle) => {
+                Vacant(VacantEntry {
+                    key: key,
+                    handle: handle,
+                    length: &mut self.length,
+                    _marker: PhantomData,
+                })
+            }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Ditto

File: src/libcollections/btree/map.rs [hunk 11]
```diff
@@ -920,8 +971,12 @@ impl<K: Ord, V> BTreeMap<K, V> {
                         left_node = edge.descend();
                         right_node = node.first_edge().descend();
                     }
-                    (Leaf(_), Leaf(_)) => { break; },
-                    _ => { unreachable!(); }
+                    (Leaf(_), Leaf(_)) => {
+                        break;
+                    }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: I think the braces and semicolon can just be killed here.

File: src/libcollections/btree/map.rs [hunk 12]
```diff
@@ -920,8 +971,12 @@ impl<K: Ord, V> BTreeMap<K, V> {
...
+                    _ => {
+                        unreachable!();
+                    }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Similarly, I think the braces and semicolon can just be killed here.

File: src/libcollections/btree/map.rs [hunk 13]
```diff
@@ -950,8 +1005,12 @@ impl<K: Ord, V> BTreeMap<K, V> {
...
                     res += dfs(edge.reborrow().descend());
                     match edge.right_kv() {
-                        Ok(right_kv) => { edge = right_kv.right_edge(); },
-                        Err(_) => { break; }
+                        Ok(right_kv) => {
+                            edge = right_kv.right_edge();
+                        }
+                        Err(_) => {
+                            break;
+                        }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Again, I think the braces and semicolon can just be killed here.

File: src/libcollections/btree/map.rs [hunk 14]
```diff
@@ -1130,14 +1193,15 @@ impl<K, V> IntoIterator for BTreeMap<K, V> {
...
+            length: len,
         }
     }
 }
 
 impl<K, V> Drop for IntoIter<K, V> {
     fn drop(&mut self) {
-        for _ in &mut *self { }
+        for _ in &mut *self {
+        }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Again, I think this just looks weird.

File: src/libcollections/btree/map.rs [hunk 15]
```diff
@@ -1865,15 +1923,17 @@ impl<'a, K: Ord, V> VacantEntry<'a, K, V> {
...
+                        Fit(_) => return unsafe { &mut *out_ptr },
+                        Split(left, k, v, right) => {
+                            ins_k = k;
+                            ins_v = v;
+                            ins_edge = right;
+                            cur_parent = left.ascend().map_err(|n| n.into_root_mut());
+                        }
                     }
-                },
+                }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Again, I don't see the value of these extra braces.

File: src/libcollections/btree/map.rs [hunk 16]
```diff
@@ -1955,14 +2011,16 @@ impl<'a, K: Ord, V> OccupiedEntry<'a, K, V> {
...
-                Stole(_) => break
+                Merged(parent) => {
+                    if parent.len() == 0 {
+                        // We must be at the root
+                        parent.into_root_mut().pop_level();
+                        break;
+                    } else {
+                        cur_node = parent.forget_type();
+                    }
+                }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Ditto

File: src/libcollections/btree/map.rs [hunk 17]
```diff
@@ -1989,10 +2045,12 @@ fn handle_underfull_node<'a, K, V>(node: NodeRef<marker::Mut<'a>,
...
-            Ok(right) => (false, right),
-            Err(parent) => {
-                return EmptyParent(parent.into_node());
+        Err(parent) => {
+            match parent.right_kv() {
+                Ok(right) => (false, right),
+                Err(parent) => {
+                    return EmptyParent(parent.into_node());
+                }
             }
```
Comment (gereeter) [inline src/libcollections/btree/map.rs]: Ditto.

**Comment by srinivasreddy** [comment]
I am not really sure how many of these comments are bugs and expected behaviours. So  over to @nrc.  And not sure changing by hand worth it.

File: src/libcollections/btree/map.rs (hunk 14 above)
Comment (nrc) [inline src/libcollections/btree/map.rs]: https://github.com/rust-lang-nursery/rustfmt/issues/1045

File: src/libcollections/btree/map.rs (hunk 3 above)
Comment (nrc) [inline src/libcollections/btree/map.rs]: https://github.com/rust-lang-nursery/rustfmt/issues/1046

**Comment by nrc** [comment]
@bors: r+

(Some of the things @gereeter pointed out are certainly sub-optimal, but can be fixed by modifying rustfmt and re-formatting)

**Pull request merged**